### DISCUSSION TITLE
How to set uploadUrl

### DISCUSSION THREAD

**Discussion by eric-hc** [sorry-cypress/sorry-cypress#301] [Q&A]
Is there a way to set the `uploadUrl` for S3? I might be missing something, but since there is `S3_READ_URL_PREFIX`, I would expect there to be one for upload URL as well.

https://github.com/sorry-cypress/sorry-cypress/blob/97f69d3b4ff0a31e75ef016496027a6210bd0017/packages/director/src/screenshots/s3/s3.ts#L51-L52

However, it looks like that is not the case, and `uploadUrl` is hardcoded

https://github.com/sorry-cypress/sorry-cypress/blob/97f69d3b4ff0a31e75ef016496027a6210bd0017/packages/director/src/screenshots/s3/s3.ts#L14

**Comment by agoldis** [comment]
@eric-hc 
you can customize the S3 upload request by modifying 
- `S3_REGION` env variable
- `S3_BUCKET`env variable
- `S3_ACL`

do you need more customization options? What's your use case?

**Reply by eric-hc** [reply]
Would this modify the `s3.amazonaws` base endpoint? We have an internal object store that uses S3 compatible REST APIs and I was wondering if I could use it.

**Reply by agoldis** [reply]
Oh, I see. 
It'll require a change to the code, not too difficult, though.

There's also minio driver; maybe you can make use of it?
